src/oscore/oscore_cbor.c

```diff
@@ -0,0 +1,430 @@
+/* -*- Mode: C; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+
+/*
+ * Copyright (c) 2018, SICS, RISE AB
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ */
+
+/**
+ * @file oscore_cbor.c
+ * @brief An implementation of the Concise Binary Object Representation (RFC).
+ *
+ * \author
+ *      Martin Gunnarsson  <martin.gunnarsson@ri.se>
+ * extended for libcoap by Peter van der Stok
+ *                         <consultancy@vanderstok.org>
+ * on request of Fairhair alliance.
+ * adapted for libcoap integration
+ *      Jon Shallow <supjps-libcoap@jpshallow.com>
+ *
+ */
+
+#include "coap3/coap_internal.h"
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+size_t
+oscore_cbor_put_nil(uint8_t **buffer, size_t *buf_size) {
```

Comment:
This screams for an inline function `util_write_byte()` that does the repeating work.

Reply:
Done this for the first 3 functions that are single byte updates.